dm/ctl/master/config.go

```diff
@@ -41,45 +40,177 @@ var (
+	if len(args) != 1 {
+		return cmd.Help()
+	}
+	name := args[0]
+	output, err := cmd.Flags().GetString("path")
+	if err != nil {
+		return err
+	}
+	return sendGetConfigRequest(pb.CfgType_SourceType, name, output)
+}
+
+// FIXME: implement this later.
+func newConfigSourceUpdateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "update <command>",
+		Short:  "update config source",
+		Hidden: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return errors.Errorf("this function will be supported later")
+		},
 	}
-	cmd.AddCommand(newExportCfgsCmd())
-	cmd.AddCommand(newImportCfgsCmd())
 	return cmd
 }
 
+func newConfigMasterCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "master [master-name]",
+		Short: "manage or show master configs",
+		RunE:  configMasterList,
+	}
+	return cmd
+}
+
+func configMasterList(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 || len(args) > 1 {
+		return cmd.Help()
+	}
+	name := args[0]
+	output, err := cmd.Flags().GetString("path")
+	if err != nil {
+		return err
+	}
+	return sendGetConfigRequest(pb.CfgType_MasterType, name, output)
+}
+
+func newConfigWorkerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "worker [worker-name]",
+		Short: "manage or show worker configs",
+		RunE:  configWorkerList,
+	}
+	return cmd
+}
+
+func configWorkerList(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 || len(args) > 1 {
+		return cmd.Help()
+	}
+	name := args[0]
+	output, err := cmd.Flags().GetString("path")
+	if err != nil {
+		return err
+	}
+	return sendGetConfigRequest(pb.CfgType_WorkerType, name, output)
+}
+
 // newExportCfgsCmd creates a exportCfg command.
 func newExportCfgsCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "export [--dir directory]",
-		Short: "Export the configurations of sources and tasks.",
+		Use:   "export",
+		Short: "Export the configurations of sources and tasks",
 		RunE:  exportCfgsFunc,
 	}
-	cmd.Flags().StringP("dir", "d", "configs", "specify the output directory, default is `./configs`")
+	cmd.Flags().String("dir", "", "specify the configs directory, default is `./configs`")
+	_ = cmd.Flags().MarkHidden("dir")
 	return cmd
 }
 
 // newImportCfgsCmd creates a importCfg command.
 func newImportCfgsCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "import [--dir directory]",
-		Short: "Import the configurations of sources and tasks.",
+		Use:   "import",
+		Short: "Import the configurations of sources and tasks",
 		RunE:  importCfgsFunc,
 	}
-	cmd.Flags().StringP("dir", "d", "configs", "specify the configs directory, default is `./configs`")
+	cmd.Flags().String("dir", "", "specify the configs directory, default is `./configs`")
+	_ = cmd.Flags().MarkHidden("dir")
 	return cmd
 }
 
 // exportCfgsFunc exports configs.
 func exportCfgsFunc(cmd *cobra.Command, args []string) error {
-	dir, err := cmd.Flags().GetString("dir")
+	filePath, err := cmd.Flags().GetString("path")
```

Comment:
I'm a bit confused by this `path` parameter. It seems like a single file path, but actually it means the path to a directory. Then why we need to support both `path` and `dir` while there is no difference

Reply:
We used to support `dir` parameter. To keep compatible with older version we support both now but mark `dir` as hidden.